File: gdscript/src/main/java/gdscript/psi/GdEnumDeclTl.java
package gdscript.psi;

import com.intellij.navigation.ItemPresentation;
import com.intellij.psi.StubBasedPsiElement;
import gdscript.index.stub.GdEnumDeclStub;
import gdscript.psi.types.GdDocumented;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import java.util.LinkedHashMap;
import java.util.List;

public interface GdEnumDeclTl extends GdTopLevelDecl, StubBasedPsiElement<GdEnumDeclStub>, GdDocumented {

  @Nullable
  GdEndStmt getEndStmt();

  @Nullable
  GdEnumDeclNmi getEnumDeclNmi();

  @NotNull
  List<GdEnumValue> getEnumValueList();

  @NotNull
  String getName();

  @NotNull
  LinkedHashMap<String, Long> getValues();

  @NotNull
  ItemPresentation getPresentation();

}

File: gdscript/src/main/java/gdscript/psi/GdEnumDeclTl.kt
package gdscript.psi

import com.intellij.navigation.ItemPresentation
import com.intellij.psi.StubBasedPsiElement
import gdscript.index.stub.GdEnumDeclStub
import gdscript.psi.types.GdDocumented

interface GdEnumDeclTl : GdTopLevelDecl, StubBasedPsiElement<GdEnumDeclStub>, GdDocumented {
    val endStmt: GdEndStmt?

    val enumDeclNmi: GdEnumDeclNmi?

    val enumValueList: List<GdEnumValue>

    fun getName(): String

    val values: LinkedHashMap<String, Long>

    fun getPresentation(): ItemPresentation
}
